### PR TITLE
Make integrations-manager reusable

### PR DESCRIPTION
**Pull request by janboll**
Making integrations manager reusable:

- add upstream as parameter to filter out integrations and fetch ref from repo
- add caller to populate_current_state to filter resource inventory on caller. makes it possible to deploy multiple upstream in one namespace
- add parameter to specify the image location from cli

This change is backwards compatible if nothing is configured.

[APPSRE-7159](https://issues.redhat.com/browse/APPSRE-7159)

### CODE REVIEW

**Review by esron — APPROVED** [review]
LGTM, but maybe you can add a section with an manual test example?

**Comment by janboll** [comment]
Logs, from running locally:


**Old implementation:**
```
/home/jboll/workspace/opensource/qontract-reconcile/venv/bin/python /home/jboll/workspace/opensource/qontract-reconcile/debugging.py --config config.dev.toml --dry-run integrations-manager
[2023-04-03 14:14:55] [INFO] [openshift_base.py:apply:373] - ['apply', 'appint-ex-01', 'app-sre', 'Deployment', 'qontract-reconcile-glitchtip-project-dsn']
[2023-04-03 14:14:55] [INFO] [openshift_base.py:apply:373] - ['apply', 'appint-ex-01', 'app-sre', 'Service', 'qontract-reconcile']
```

**New implementation, no upstream:**
```
/home/jboll/workspace/opensource/qontract-reconcile/venv/bin/python /home/jboll/workspace/opensource/qontract-reconcile/debugging.py --config config.dev.toml integrations-manager
[2023-04-03 14:22:14] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
[2023-04-03 14:22:15] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
```

**New implementation, upstream set:**
```
/home/jboll/workspace/opensource/qontract-reconcile/venv/bin/python /home/jboll/workspace/opensource/qontract-reconcile/debugging.py --config config.dev.toml integrations-manager
[2023-04-03 14:24:54] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
[2023-04-03 14:24:54] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
[2023-04-03 14:25:07] [INFO] [openshift_base.py:apply:380] - ['apply', 'appint-ex-01', 'app-sre', 'Service', 'qontract-reconcile']
[2023-04-03 14:25:07] [INFO] [openshift_base.py:apply:380] - ['apply', 'appint-ex-01', 'app-sre', 'Deployment', 'qontract-reconcile-glitchtip-project-dsn']
```
**Second run:**
```
/home/jboll/workspace/opensource/qontract-reconcile/venv/bin/python /home/jboll/workspace/opensource/qontract-reconcile/debugging.py --config config.dev.toml integrations-manager
[2023-04-03 14:25:25] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
[2023-04-03 14:25:25] [INFO] [gql.py:init_from_config:292] - using gql endpoint http://localhost:4000/graphqlsha/2f06c7dd9eda5871a77a52968ebb1ce6f08b7ac885cbdb95c0f0acd1b8440887
```

File: reconcile/integrations_manager.py [hunk 1]
```diff
@@ -369,15 +396,28 @@ def run(
...
+        "thread_pool_size": thread_pool_size,
+        "integration": QONTRACT_INTEGRATION,
+        "integration_version": QONTRACT_INTEGRATION_VERSION,
+        "override_managed_types": ["Deployment", "StatefulSet", "CronJob", "Service"],
+        "internal": internal,
+        "use_jump_host": use_jump_host,
+    }
+
+    if not image:
+        image = "quay.io/app-sre/qontract-reconcile"
```
Comment (kwilczynski) [inline reconcile/integrations_manager.py:410]: Would it make sense to turn this into a constant?

File: reconcile/integrations_manager.py [hunk 2]
```diff
@@ -369,15 +396,28 @@ def run(
...
+
+    if upstream:
+        use_upstream = True
+        fetch_args["caller"] = upstream
+    else:
+        # Not set to fetch_args on purpose, fallback for cases where caller is not yet set
+        use_upstream = False
+        upstream = "https://github.com/app-sre/qontract-reconcile"
```
Comment (kwilczynski) [inline reconcile/integrations_manager.py:418]: Same as above. Would it make sense to turn this into a constant?

**Review by kwilczynski — APPROVED** [review]
@janboll, looks good! Thank you!

File: reconcile/integrations_manager.py [hunk 3]
```diff
@@ -369,15 +399,28 @@ def run(
...
+        "thread_pool_size": thread_pool_size,
+        "integration": QONTRACT_INTEGRATION,
+        "integration_version": QONTRACT_INTEGRATION_VERSION,
+        "override_managed_types": ["Deployment", "StatefulSet", "CronJob", "Service"],
+        "internal": internal,
+        "use_jump_host": use_jump_host,
+    }
+
+    if not image:
+        image = IMAGE_DEFAULT
```
Comment (geoberle) [inline reconcile/integrations_manager.py:413]: do we need defaults? what prevents us from expecting values at all times?

**Review by geoberle — APPROVED** [review]
only nits. i love how you implemented this by adding details at the right places! 🙇

File: reconcile/integrations_manager.py (hunk 3 above)
Comment (janboll) [reply reconcile/integrations_manager.py:413]: Yes, to minimize changes required during the rollout for Fedramp

Comment (geoberle) [reply reconcile/integrations_manager.py:413]: ah understood. lets revisite once this has been rolled out everywhere

**Pull request merged**